zebra-chain/src/value_balance.rs

```diff
@@ -0,0 +1,293 @@
+//! A type that can hold the four types of Zcash value pools.
+
+use crate::amount::{Amount, Constraint, Error, NegativeAllowed, NonNegative};
```

Comment:
Let's fix the clippy warnings, and any of the optional changes you'd like to make.

Then I think we're good to merge!